### PR TITLE
socks proxy rebase

### PR DESCRIPTION
**Pull request by ndew623**
A rebase that brings the socks_proxy branch up to date with the master branch for easier future development. And this time the base is k9mail/master and there are 6 commits, not over 1000.

### CODE REVIEW

**Comment by Valodim** [comment]
Nice work! The unit tests seem to be failing now due to a syntax error, do you want to look into this?

**Comment by ndew623** [comment]
Thanks! I believe I have solved the issue with the tests.

File: k9mail/src/main/java/com/fsck/k9/K9.java [hunk 1]
```diff
@@ -1390,11 +1401,11 @@ public static boolean isMessageViewSpamActionVisible() {
     public static void setMessageViewSpamActionVisible(boolean visible) {
         sMessageViewSpamActionVisible = visible;
     }
-
     public static int getPgpInlineDialogCounter() {
         return sPgpInlineDialogCounter;
     }
 
+
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/K9.java]: unnecessary whitespace changes

File: k9mail-library/src/main/java/com/fsck/k9/mail/ssl/DefaultTrustedSocketFactory.java [hunk 2]
```diff
@@ -203,6 +223,11 @@ public Socket createSocket(Socket socket, String host, int port, String clientCe
 
         setSniHost(socketFactory, sslSocket, host);
 
+        if (!proxySettings.enabled && socket == null) {
```
Comment (Valodim) [inline k9mail-library/src/main/java/com/fsck/k9/mail/ssl/DefaultTrustedSocketFactory.java]: I don't understand this change, and how the `socket.connect` methods below are all removed. Does the method return a connected socket now, or what?

**Comment by Valodim** [comment]
Thanks for picking this up! I think we should try to merge this asap, to avoid collecting bitdust again. If we keep this feature flagged, it should be simple to ensure nothing breaks.

File: k9mail-library/src/main/java/com/fsck/k9/mail/ssl/DefaultTrustedSocketFactory.java (hunk 2 above)
Comment (philipwhiuk) [inline k9mail-library/src/main/java/com/fsck/k9/mail/ssl/DefaultTrustedSocketFactory.java]: Yes I think it does. Because the proxied case has to return a connected socket, all the methods do. I think this is okay but the method name should indicate that this is the case.

**Comment by cketti** [comment]
The `socks_proxy` branch hasn't been merged because it needs additional work done. Not because it just needs rebasing. Closing this PR.

If someone wants to work on this it's probably easier to start fresh. I don't want any of the remote stores to have to know about a proxy server. We should be able to pass a Socket factory to the remote stores. All the SOCKS proxy handling code could live in the factory class.
I think Valodim is right in that we should use BuildConfig.DEBUG as feature flag.
